### PR TITLE
[SYCL][E2E] Remove xfail from Joint Matrix prefetch tests

### PR DESCRIPTION
**Pull request by dkhaldi**

### CODE REVIEW

**Review by uditagarwal97 — APPROVED** [review]

File: sycl/test-e2e/Matrix/joint_matrix_bf16_fill_k_cache_prefetch.cpp [hunk 1]
```diff
@@ -16,11 +16,5 @@
 
 // -ffp-model=precise is added to not depend on compiler defaults.
 
-// This test will be adjusted to run on the CPU
-// XFAIL:cpu
```
Comment (uditagarwal97) [inline sycl/test-e2e/Matrix/joint_matrix_bf16_fill_k_cache_prefetch.cpp:20]: Is the test passing on CPU as well?

**Review by YuriPlyakhin — APPROVED** [review]
LGTM

File: sycl/test-e2e/Matrix/joint_matrix_bf16_fill_k_cache_prefetch.cpp (hunk 1 above)
Comment (dkhaldi) [reply sycl/test-e2e/Matrix/joint_matrix_bf16_fill_k_cache_prefetch.cpp:20]: Yes, that is a separate issue but the test is also xpasssing on the CPU

**Review by aelovikov-intel — APPROVED** [review]

**Comment by dkhaldi** [comment]
Jenkins/Precommit is failing because it is considering XFAIL-TRACKER: CMPLRLLVM-62790 as an IP leak which it should not.
@uditagarwal97  said we can ignore this.
@intel/llvm-gatekeepers, can you help merge this?

**Pull request merged**